x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.tsx

```diff
@@ -26,26 +29,87 @@ import {
 } from '../../state_syncing/global_sync';
 import { AppStateManager } from '../../state_syncing/app_state_manager';
 import { startAppStateSyncing } from '../../state_syncing/app_sync';
-import { esFilters } from '../../../../../../../src/plugins/data/public';
+import {
+  esFilters,
+  Filter,
+  Query,
+  TimeRange,
+  IndexPattern,
+  SavedQuery,
+  QueryStateChange,
+  QueryState,
+} from '../../../../../../../src/plugins/data/public';
 import { MapContainer } from '../../../connected_components/map_container';
 import { getIndexPatternsFromIds } from '../../../index_pattern_util';
 import { getTopNavConfig } from './top_nav_config';
 import { getBreadcrumbs, unsavedChangesWarning } from './get_breadcrumbs';
+import {
+  LayerDescriptor,
+  MapRefreshConfig,
+  MapCenterAndZoom,
+  MapQuery,
+} from '../../../../common/descriptor_types';
+import { MapSettings } from '../../../reducers/map';
+import { ISavedGisMap } from '../../bootstrap/services/saved_gis_map';
+
+interface Props {
+  savedMap: ISavedGisMap;
+  onAppLeave: AppMountParameters['onAppLeave'];
+  stateTransfer: EmbeddableStateTransfer;
+  originatingApp?: string;
+  layerListConfigOnly: LayerDescriptor[];
+  replaceLayerList: (layerList: LayerDescriptor[]) => void;
+  filters: Filter[];
+  isFullScreen: boolean;
+  isOpenSettingsDisabled: boolean;
+  enableFullScreen: () => void;
+  openMapSettings: () => void;
+  inspectorAdapters: Adapters;
+  nextIndexPatternIds: string[];
+  dispatchSetQuery: ({
+    forceRefresh,
+    filters,
+    query,
+    timeFilters,
+  }: {
+    filters?: Filter[];
+    query?: Query;
+    timeFilters?: TimeRange;
+    forceRefresh?: boolean;
+  }) => void;
+  timeFilters: TimeRange;
+  refreshConfig: MapRefreshConfig;
+  setRefreshConfig: (refreshConfig: MapRefreshConfig) => void;
+  isSaveDisabled: boolean;
+  clearUi: () => void;
+  setGotoWithCenter: (latLonZoom: MapCenterAndZoom) => void;
+  setMapSettings: (mapSettings: MapSettings) => void;
+  setIsLayerTOCOpen: (isLayerTOCOpen: boolean) => void;
+  setOpenTOCDetails: (openTOCDetails: string[]) => void;
+  query: MapQuery | undefined;
+}
 
-export class MapsAppView extends React.Component {
-  _globalSyncUnsubscribe = null;
-  _globalSyncChangeMonitorSubscription = null;
-  _appSyncUnsubscribe = null;
+interface State {
+  initialized: boolean;
+  initialLayerListConfig?: LayerDescriptor[];
+  indexPatterns: IndexPattern[];
+  savedQuery?: SavedQuery;
+  originatingApp?: string;
+}
+
+export class MapsAppView extends React.Component<Props, State> {
+  _globalSyncUnsubscribe: (() => void) | null = null;
+  _globalSyncChangeMonitorSubscription: Subscription | null = null;
+  _appSyncUnsubscribe: (() => void) | null = null;
   _appStateManager = new AppStateManager();
-  _prevIndexPatternIds = null;
+  _prevIndexPatternIds: string[] | null = null;
+  _isMounted?: boolean;
```

Comment:
Instead of making optional, just init to false

Reply:
Fixed this.